code/game/objects/structures/morgue.dm

```diff
@@ -203,6 +217,58 @@ GLOBAL_LIST_EMPTY(bodycontainers) //Let them act as spawnpoints for revenants an
 	name = "morgue memo"
 	info = "<font size='2'>Since this station's medbay never seems to fail to be staffed by the mindless monkeys meant for genetics experiments, I'm leaving a reminder here for anyone handling the pile of cadavers the quacks are sure to leave.</font><BR><BR><font size='4'><font color=red>Red lights mean there's a plain ol' dead body inside.</font><BR><BR><font color=orange>Yellow lights mean there's non-body objects inside.</font><BR><font size='2'>Probably stuff pried off a corpse someone grabbed, or if you're lucky it's stashed booze.</font><BR><BR><font color=green>Green lights mean the morgue system detects the body may be able to be brought back to life.</font></font><BR><font size='2'>I don't know how that works, but keep it away from the kitchen and go yell at the geneticists.</font><BR><BR>- CentCom medical inspector"
 
+/*
+ * Morgue (Extraction Variant)
+ */
+
+/obj/structure/bodycontainer/morgue/extraction
+	name = "headstone"
+	desc = "A sleek stone container, etched apon it's surface is a glowing gold text. Along it's sides are elaborate grooves and creases."
+	icon_state = "extract_morgue_empty"
+	dir = EAST
+	beeper = TRUE
+	beep_cooldown = 50
+	next_beep = 0
+
+/obj/structure/bodycontainer/morgue/extraction/Initialize()
+	. = ..()
+	connected = new/obj/structure/tray/e_tray(src)
+	connected.connected = src
+
+/obj/structure/bodycontainer/morgue/extraction/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>The speaker is [beeper ? "enabled" : "disabled"]. Alt-click to toggle it.</span>"
+
+/obj/structure/bodycontainer/morgue/extraction/AltClick(mob/user)
```

Comment:
Shouldn't all this stuff be moved to the parent type?

Reply:
Done!